### PR TITLE
Readying repo for migration to GHA

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
This PR will get the repo ready to be moved to GitHub Actions.

- Creates two non-destructive GHA workflows, for on `push` and `pull_request` 
- Removes Packer.  Deemed unnecessary and clunky.  We could look to replace with something else?
- Added lifecycle protection to peering connection

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        working-directory: .
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.ACTIONS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ACTIONS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: eu-west-2
+
+      - name: "Terraform Format"
+        id: fmt
+        run: terraform fmt
+        continue-on-error: true
```
Comment (dan-hill2802) [inline .github/workflows/pull-request.yml:41]: I guess this isn't technically needed for a pipeline plan/apply

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: "Push Checks"
+on:
+  push:
+    paths:
+      - "terraform/**"
```
Comment (dan-hill2802) [inline .github/workflows/push.yml:5]: should also include this `push.yml` file

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 0.12.19
+
+      - name: "Terraform Format"
+        id: fmt
+        run: terraform fmt
+        continue-on-error: true
```
Comment (dan-hill2802) [inline .github/workflows/push.yml:23]: is the plan to add `apply` later? If so, maybe we could have the job put in now but just have it do a `plan` for now, just so we can see everything is in working order

File: bootstrap_terraform.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 
 
 def main():
+    region_name = os.environ.get('AWS_DEFAULT_REGION')
```
Comment (dan-hill2802) [inline bootstrap_terraform.py:15]: this is never used, should it be?

File: terraform/deploy/terraform.tfvars.j2 [hunk 5]
```diff
@@ -35,8 +35,3 @@ github_vpc = {
 {%- for k, v in ucfs['github']['vpc'].items() %}
   {{ k }} = "{{ v }}"{% endfor %}
 }
-
-packer_egress_test_zip = {
-  base_path = "../../../packer-egress-test-release"
-  version   = "0.0.12"
-}
```
Comment (dan-hill2802) [inline terraform/deploy/terraform.tfvars.j2:42]: don't we still need this for running tests in the pipeline?

File: terraform/deploy/test_lambda.tf [hunk 6]
```diff
@@ -1,47 +0,0 @@
-data "aws_iam_role" "packer_egress_test" {
```
Comment (dan-hill2802) [inline terraform/deploy/test_lambda.tf:1]: why are we removing this Lambda?

File: .github/workflows/push.yml (hunk 2 above)
Comment (ChrisScottThomas) [reply .github/workflows/push.yml:5]: I only want the workflow to run against changes in Terraform?

File: .github/workflows/push.yml (hunk 3 above)
Comment (ChrisScottThomas) [reply .github/workflows/push.yml:23]: yes there will be a job for merging to master that runs the apply.  We're not there yet.

File: terraform/deploy/terraform.tfvars.j2 (hunk 5 above)
Comment (ChrisScottThomas) [reply terraform/deploy/terraform.tfvars.j2:42]: It requires the zip file to be present.  The GHA is ephemeral and does not have the zip file.  
It's only used when creating AWS-Concourse, to run a test in Concourse.  We're no longer using Concourse

File: terraform/deploy/test_lambda.tf (hunk 6 above)
Comment (ChrisScottThomas) [reply terraform/deploy/test_lambda.tf:1]: Same as above.

File: .github/workflows/push.yml (hunk 2 above)
Comment (dan-hill2802) [reply .github/workflows/push.yml:5]: It should also run when the pipeline changes, like adding a TF_VAR environment variable or changing the workspace

Comment (ChrisScottThomas) [reply .github/workflows/push.yml:5]: All it does is Terraform FMT, on Terraform files. 🤔

**Review by mattburgess — APPROVED** [review]

**Pull request merged**